### PR TITLE
extensions: update docs for the air-gapped instances

### PR DESCRIPTION
**Pull request by taras-yemets**
Closes https://github.com/sourcegraph/sourcegraph/issues/36851

Context: [RFC 681](https://docs.google.com/document/d/1ymCMW9TBeirdqjTPx0zmYfNc5v-zYKCwLF2bxZJzXiA/edit#)

## Test plan
Visually tested by running the doc site locally.

### CODE REVIEW

File: doc/code_intelligence/how-to/enable_code_intel_on_air_gapped_instances.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Enable code intelligence on the air-gapped instances
+
+Sourcegraph code intelligence [is implemented on top of extensions](../../dev/background-information/codeintel/extensions.md). Code-intel extensions are [enabled by default](../../extensions/usage.md#default-extensions). Sourcegraph extensions are fetched from [sourcegraph.com extensions registry](https://sourcegraph.com/extensions), so to make use of them the Sourcegraph instance should have Internet access and properly configured [`extensions.remoteRegistry`](../../admin/extensions/index.md#use-extensions-from-sourcegraphcom-or-disable-remote-extensions) site config setting.
```
Comment (abeatrix) [inline doc/code_intelligence/how-to/enable_code_intel_on_air_gapped_instances.md:3]: > Code-intel extensions are enabled by default. 

Are code-intel extensions = programming language extensions, or are they different?

Comment (abeatrix) [inline doc/code_intelligence/how-to/enable_code_intel_on_air_gapped_instances.md:3]:
```suggestion
Sourcegraph code intelligence [is implemented on top of extensions](../../dev/background-information/codeintel/extensions.md). Code-intel extensions are [enabled by default](../../extensions/usage.md#default-extensions). Sourcegraph extensions are fetched from [sourcegraph.com extensions registry](https://sourcegraph.com/extensions), so to make use of them the Sourcegraph instance should have Internet access and properly configured [`extensions.remoteRegistry`](../../admin/extensions/index.md#use-extensions-from-sourcegraphcom-or-disable-remote-extensions) site config setting. Using a proxy to access the [Sourcegraph extensions registry](https://sourcegraph.com/extensions) is not supported.
```

**Review by abeatrix — APPROVED** [review]
This is super helpful! I am tagging the support team for reviews as well since they will be the ones using the docs 

CC @topebali @airamare01 since you both have recently worked with air-gapped customers

File: doc/code_intelligence/how-to/enable_code_intel_on_air_gapped_instances.md (hunk 1 above)
Comment (taras-yemets) [reply doc/code_intelligence/how-to/enable_code_intel_on_air_gapped_instances.md:3]: All code-intel extensions belong to the "programming languages" category, but the latter may contain other extensions too.
I understand code intel extensions as the ones generated from [code-intel-extensions](https://github.com/sourcegraph/code-intel-extensions) repo.

**Review by chrismwendt — APPROVED** [review]

**Pull request merged**